core/services/eal/blockchain_client.go

```diff
@@ -0,0 +1,124 @@
+package eal
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/pkg/errors"
+
+	"github.com/ethereum/go-ethereum"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	eallib "github.com/smartcontractkit/capital-markets-projects/lib/services/eal"
+	txmgrcommon "github.com/smartcontractkit/chainlink/v2/common/txmgr"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/client"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/config"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/txmgr"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/job"
+	"github.com/smartcontractkit/chainlink/v2/core/services/keystore"
+	"github.com/smartcontractkit/chainlink/v2/core/services/pg"
+)
+
+var _ eallib.BlockchainClientInterface = &BlockchainClient{}
+
+type BlockchainClient struct {
+	lggr    logger.Logger
+	txm     txmgr.TxManager
+	gethks  keystore.Eth
+	spec    job.EALSpec
+	chainID uint64
+	cfg     config.EVM
+	client  client.Client
+}
+
+func NewBlockchainClient(
+	lggr logger.Logger,
+	txm txmgr.TxManager,
+	gethks keystore.Eth,
+	spec job.EALSpec,
+	chainID uint64,
+) (*BlockchainClient, error) {
+	return &BlockchainClient{
+		lggr:    lggr,
+		txm:     txm,
+		gethks:  gethks,
+		spec:    spec,
+		chainID: chainID,
+	}, nil
+}
+
+func (c *BlockchainClient) SimulateAndCreateTransaction(
```

Comment:
Just wondering but would it be better to have more primitive methods as part of the blockchain client? e.g 

```golang
SimulateTransaction(ctx context.Context, toAddress gethCommon.Address, payload []byte) error
CreateTransaction(ctx context.Context, toAddress gethCommon.Address, payload []byte) error
```

This way the EAL lib can simulate without having to create a transaction necessarily. It can also do both, since chaining the calls is trivial at the call site.

Reply:
Note that the EAL will sometimes have to do plain simulations (eth_calls) without actually submitting a tx.

Reply:
Perhaps you were planning on having that as a separate method though.

Reply:
yeah I think it would be useful to breakdown this into multiple methods. will update